src/transformers/models/t5/modeling_t5.py

```diff
@@ -434,21 +454,41 @@ def _relative_position_bucket(relative_position, bidirectional=True, num_buckets
         relative_buckets += torch.where(is_small, relative_position, relative_position_if_large)
         return relative_buckets
 
-    def compute_bias(self, query_length, key_length, device=None):
+    def compute_bias_for_position_embedding_name(self, position_embedding_name, position_ids, query_length, key_length, device=None):
+        """Compute binned relative position bias"""
+        if position_embedding_name == POSITION_EMBEDDING_T5_RELATIVE:
+            relative_attention_bias = self.relative_attention_bias
+            num_buckets=self.relative_attention_num_buckets
+            max_distance=self.relative_attention_max_distance                
+        else:
+            relative_attention_bias = self.relative_attention_bias_dict[position_embedding_name]
+            num_buckets =self.relative_position_embedding_definitions[position_embedding_name].get("num_buckets", self.relative_attention_num_buckets)
+            max_distance=self.relative_position_embedding_definitions[position_embedding_name].get("max_distance", self.relative_attention_max_distance)
+
+        return self.compute_bias(relative_attention_bias=relative_attention_bias, position_ids=position_ids, num_buckets=num_buckets, max_distance=max_distance, query_length=query_length, key_length=key_length, device=device)
+
+    def compute_bias(self, relative_attention_bias, position_ids, num_buckets, max_distance, query_length, key_length, device=None):
         """Compute binned relative position bias"""
         if device is None:
-            device = self.relative_attention_bias.weight.device
-        context_position = torch.arange(query_length, dtype=torch.long, device=device)[:, None]
-        memory_position = torch.arange(key_length, dtype=torch.long, device=device)[None, :]
+            device = relative_attention_bias.weight.device
+        if position_ids is not None:
+            context_position = position_ids.unsqueeze(2)
```

Comment:
what is the shape before unsqueeze?

Reply:
When position_ids are given we have additional batch dimension: shape =(num_batches, key_and_query_length)
I added this in a comment:
```
context_position = position_ids.unsqueeze(2) # shape (num_batches, key_and_query_lnegth) ->  (num_batches, key_and_query_lnegth, 1)
 memory_position = position_ids.unsqueeze(1) # shape (num_batches, key_and_query_lnegth) ->  (num_batches, 1, key_and_query_lnegth)
```